risc0/zkvm/src/prove/mod.rs

```diff
@@ -127,6 +145,40 @@ impl<'a> Default for ProverOpts<'a> {
 }
 
 /// Manages communication with and execution of a zkVM [Program]
+///
+/// # Usage
+/// A [Prover] is constructed from the ELF code and an Image ID generated from
+/// the guest code to be proven (see
+/// [risc0_build](https://docs.rs/risc0-build/latest/risc0_build/) for more
+/// information about how these are generated). Use [Prover::new] if you want
+/// the default [ProverOpts], or [Prover::new_with_opts] to use custom options.
+/// ```ignore
+/// // In real code, the ELF & Image ID would be generated by risc0 build scripts from guest code
+/// use methods::{EXAMPLE_ELF, EXAMPLE_ID};
+/// use risc0_zkvm::Prover;
+///
+/// let mut prover = Prover::new(&EXAMPLE_ELF, EXAMPLE_ID)?;
+/// ```
+/// Provers should essentially always be mutable so that their [Prover::run]
+/// method may be called.
+///
+/// Input data can be passed to the Prover with [Prover::add_input_u32_slice]
+/// (or [Prover::add_input_u8_slice]). After all inputs have been added, call
+/// [Prover::run] to execute the guest code and produce a [Receipt] proving
+/// execution.
+/// ```ignore
+/// prover.add_input_u32_slice(&to_vec(&input)?);
+/// let receipt = prover.run()?;
+/// ```
+/// After running the prover, publicly proven results can be accessed from the
+/// [Receipt], while private outputs can be accessed using
+/// [Prover::get_output_u32_vec] (or [Prover::get_output_u8_slice]):
+/// ```ignore
+/// let receipt = prover.run()?;
+/// let slice = prover.get_output_u32_vec()?;
+/// let private_output: OutputType = risc0_zkvm::serde::from_slice(&slice)?;
+/// let proven_result: ResultType = risc0_zkvm::serde::from_slice(&receipt.journal)?;
+/// ```
```

Comment:
It's important to mention the `pub cycles` member that can be used by the host to determine how many cycles the zkvm executed. It's not part of the receipt because it should be private to only the host.

Reply:
Also missing is docs for the `pub fn default_hal()`